akka-stream/src/main/scala/akka/stream/impl/PhasedFusingActorMaterializer.scala

```diff
@@ -382,22 +380,21 @@ case class PhasedFusingActorMaterializer(
       subflowFuser,
       initialAttributes,
       PhasedFusingActorMaterializer.DefaultPhase,
-      PhasedFusingActorMaterializer.DefaultPhases
-    )
+      PhasedFusingActorMaterializer.DefaultPhases)
   }
 
   def materialize[Mat](
     graph:             Graph[ClosedShape, Mat],
     subflowFuser:      GraphInterpreterShell ⇒ ActorRef,
     initialAttributes: Attributes,
     defaultPhase:      Phase[Any],
-    phases:            Map[IslandTag, Phase[Any]]
-  ): Mat = {
-    val islandTracking = new IslandTracking(phases, settings, defaultPhase, this)
+    phases:            Map[IslandTag, Phase[Any]]): Mat = {
+    val islandTracking = new IslandTracking(phases, settings, defaultPhase, this, stageNamePrefix = createFlowName() + "-")
 
     var current: Traversal = graph.traversalBuilder.traversal
 
     val attributesStack = new java.util.ArrayDeque[Attributes](8)
+    println(s"# initialAttributes $initialAttributes") // FIXME
     attributesStack.addLast(initialAttributes)
```

Comment:
Can you paste the snippet? Did you use `named`, `withAttributes` or `addAttributes`?